keeper-gradle-plugin/src/main/java/com/slack/keeper/KeeperPlugin.kt

```diff
@@ -170,16 +170,19 @@ public class KeeperPlugin : Plugin<Project> {
       }
     }
 
-    val androidJarFileProvider = provider {
-      val compileSdkVersion = appExtension.compileSdkVersion
-          ?: error("No compileSdkVersion found")
-      File("${appExtension.sdkDirectory}/platforms/${compileSdkVersion}/android.jar").also {
-        check(it.exists()) {
-          "No android.jar found! Expected to find it at: $it"
-        }
-      }
+    val androidEmbeddedJarRegularFileProvider: (String) -> Provider<RegularFile> = { path: String ->
+        layout.file(provider {
+            val compileSdkVersion = appExtension.compileSdkVersion
+                ?: error("No compileSdkVersion found")
+            File("${appExtension.sdkDirectory}/platforms/${compileSdkVersion}/${path}").also {
+                check(it.exists()) {
+                    "No $path found! Expected to find it at: $it"
+                }
+            }
+        })
     }
-    val androidJarRegularFileProvider = layout.file(androidJarFileProvider)
+    val androidJarRegularFileProvider = androidEmbeddedJarRegularFileProvider("android.jar")
+    val androidTestJarRegularFileProvider = androidEmbeddedJarRegularFileProvider("optional/android.test.base.jar")
```

Comment:
Could you document the details on this a bit? Just comments inline is fine, but I'm not totally familiar with what this `android.test.base.jar` is for

Reply:
Me too, that was one of the provided jar I was advised to consider in my commands in order to reduce the number of undeclared classes. Sadly I don't know much more about it, including if it's always available or not 😕 

Could you give us some more insights @sgjesse ?

See https://issuetracker.google.com/issues/173435379 ticket for the context.

Reply:
The setup for Android tests with JUnit classes is somewhat complicated. Take a look at https://developer.android.com/training/testing/set-up-project#add-gradle-dependencies (bullet 4) and the following section https://developer.android.com/training/testing/set-up-project#add-manifest-declarations. The classes provided by the `optional` jars are listed at https://developer.android.com/training/testing/set-up-project#junit-based-libs.

The reason for this is that historically parts of JUnit has been built into the Android platform, and initially provided through `android.jar`. The splitting out of these classes and the manifest flags to enable them at runtime is a way to disentangle this again, and was introduced from API level 28 (Android 9).

In the same way the Apache HTTP library was already split out from API level 23 (Android 6.0) (https://developer.android.com/about/versions/marshmallow/android-6.0-changes#behavior-apache-http-client).

Reply:
So I guess this file could be missing for applications not targeting the API level 28.
I made this file resolution optional.